Sample Applications/ExpenseIt/ExpenseItDemo/CreateExpenseReportDialogBox.xaml

```diff
@@ -20,32 +20,49 @@
             </ResourceDictionary.MergedDictionaries>
 
             <DataTemplate x:Key="ExpenseTemplate">
-                <Grid>
-                    <Grid.ColumnDefinitions>
-                        <ColumnDefinition Width="33*" />
-                        <ColumnDefinition Width="33*" />
-                        <ColumnDefinition Width="33*" />
-                    </Grid.ColumnDefinitions>
-                    <TextBox Text="{Binding Path=Type}" Grid.Column="0" />
-                    <TextBox Text="{Binding Path=Description}" Grid.Column="1" />
-                    <TextBox Grid.Column="2" TextAlignment="Right">
-                        <TextBox.Text>
-                            <Binding Path="Cost" UpdateSourceTrigger="PropertyChanged">
-                                <!-- SECURITY: Cost must be an int -->
-                                <Binding.ValidationRules>
-                                    <localValidation:NumberValidationRule />
-                                </Binding.ValidationRules>
-                            </Binding>
-                        </TextBox.Text>
-                    </TextBox>
-                </Grid>
+
+                <DataGrid>
+                    <DataGrid.Columns>
+                        <DataGridTemplateColumn>
```

Comment:
Why not just [DataGridTextColumn](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.datagridtextcolumn?view=netframework-4.8)?

Reply:
fixed